File: src/code-samples/deepagents/tools-mcp.py
"""Deep Agents tools page: MCP example."""

# :remove-start:
from deepagents import create_deep_agent

test_agent = create_deep_agent(model="anthropic:claude-sonnet-4-6", tools=[])
assert test_agent is not None
print("✓ tools-mcp sample wiring validated")
raise SystemExit(0)
# :remove-end:

# :snippet-start: tools-mcp-py
import asyncio
from langchain_mcp_adapters.client import MultiServerMCPClient
from deepagents import create_deep_agent


async def main():
    client = MultiServerMCPClient(
        {
            "my_server": {
                "transport": "http",
                "url": "http://localhost:8000/mcp",
            }
        }
    )
    tools = await client.get_tools()

    agent = create_deep_agent(
        model="openai:gpt-5.5",
        tools=tools,
    )

    result = await agent.ainvoke(
        {"messages": [{"role": "user", "content": "Use the MCP server to help me."}]},
        config={"configurable": {"thread_id": "1"}},
    )


asyncio.run(main())
# :snippet-end:

File: src/code-samples/deepagents/tools-mcp.ts
// :remove-start:
import { createDeepAgent } from "deepagents";

const testAgent = await createDeepAgent({
  model: "anthropic:claude-sonnet-4-6",
  tools: [],
});
if (!testAgent) throw new Error("agent not created");
console.log("✓ tools-mcp sample wiring validated");
process.exit(0);
// :remove-end:

// :snippet-start: tools-mcp-js
import { createDeepAgent } from "deepagents";

const { MultiServerMCPClient } = await import("@langchain/mcp-adapters");

const client = new MultiServerMCPClient({
  my_server: {
    transport: "http",
    url: "http://localhost:8000/mcp",
  },
});

const tools = await client.getTools();

const agent = await createDeepAgent({
  model: "openai:gpt-5.5",
  tools,
});

const result = await agent.invoke({
  messages: [{ role: "user", content: "Use the MCP server to help me." }],
});
// :snippet-end:
